### PR TITLE
Update appveyor to java17

### PR DESCRIPTION
**Pull request by GiorgioBrux**
Following TechFortress/GriefPrevention#1663 as a reference, fix appveyor and update to java 17.
This will produce a build, assuming #292 #296 are merged.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -3,33 +3,25 @@ skip_tags: true
 clone_depth: 2
 environment:
   matrix:
-  - JAVA_HOME: C:\Program Files\Java\jdk1.8.0
+    - appveyor_build_worker_image: Visual Studio 2022
+      JAVA_HOME: C:\Program Files\Java\jdk17
 branches:
   only:
-  - master
```
Comment (RoboMWM) [inline appveyor.yml:9]: this is valid yaml syntax

File: appveyor.yml [hunk 2]
```diff
@@ -3,33 +3,25 @@ skip_tags: true
...
+    - master
   except:
-  - gh-pages
+    - gh-pages
 os: Windows Server 2012
```
Comment (RoboMWM) [inline appveyor.yml:13]: unsure if this needs to be removed?

**Review by RoboMWM — COMMENTED** [review]
Unsure why the formatting changes. I'll get it pulled tomorrow tho if you decide to undo these or not I guess.

But otherwise thank you, very much appreciated!

File: appveyor.yml (hunk 2 above)
Comment (GiorgioBrux) [reply appveyor.yml:11]: yep, removed.

File: appveyor.yml (hunk 1 above)
Comment (GiorgioBrux) [reply appveyor.yml:9]: yes. and?

**Comment by GiorgioBrux** [comment]
Sorry for the formatting changes, fucked something up on my end. Force-pushed a commit with only actual changes.

File: appveyor.yml (hunk 1 above)
Comment (RoboMWM) [reply appveyor.yml:9]: unnecessary changes make it harder to read the diff

**Review by RoboMWM — APPROVED** [review]

**Pull request merged**